Sources/Server/Parser.swift

```diff
@@ -1,32 +1,68 @@
 import Foundation
 
 public class Parser {
-    
-   public init(){}
-
-   public func parseRequest(request: String) -> Request {
-        let requestComponents = request.components(separatedBy: " ")
-            if (requestComponents[1].contains("?")) {
-                let parsedPathWithQueries = parsePath(path: requestComponents[1])
-                let queries = parseAllQueries(parsedQuery: parsedPathWithQueries[1])
-                return Request(method: requestComponents[0], path: parsedPathWithQueries[0], queries: queries) 
-            }
-        return Request(method: requestComponents[0], path: requestComponents[1])
-   }
-
-   private func parsePath(path: String) -> Array<String> {
+
+    public init(){}
+
+    public func parseRequest(request: String) -> Request {
+        var (requestComponents, requestHeaders) = parseRequest(request: request)
+        let headers = parseHeaders(requestHeaders: requestHeaders)
+        if (requestComponents[1].contains("?")) {
+            let parsedPathWithQueries = parsePath(path: String(requestComponents[1]))
+            let queries = parseAllQueries(parsedQuery: parsedPathWithQueries[1])
+            return Request(method: String(requestComponents[0]), path: parsedPathWithQueries[0], queries: queries) 
+        } else if (headers.keys.contains("Cookie")) {
+            let cookie = parseCookie(headers: headers)
+            return Request(method: String(requestComponents[0]), path: String(requestComponents[1]), cookie: cookie["type"]!) 
+        } else {
+            return Request(method: String(requestComponents[0]), path: String(requestComponents[1]))
+        }
+    }
+
+    private func parseRequest(request: String) -> (Array<Substring>, Array<String>) {
```

Comment:
Although Swift seems to allow functions with the same name and arguments but different return types, I think I'd prefer each function had a unique name to better be able to distinguish them. Thoughts, @tommcgee?

Also, is `Array<Substring>` legit?

Reply:
Whoops! My mistake. Yes, Array<Substring> is a thing. I could convert it to an Array<String>. I'll go ahead and do that.